cf/commands/servicekey/service_key_test.go

```diff
@@ -0,0 +1,127 @@
+package servicekey_test
+
+import (
+	"github.com/cloudfoundry/cli/cf/configuration/core_config"
+	"github.com/cloudfoundry/cli/cf/models"
+	"github.com/cloudfoundry/cli/generic"
+
+	testapi "github.com/cloudfoundry/cli/cf/api/fakes"
+	testcmd "github.com/cloudfoundry/cli/testhelpers/commands"
+	testconfig "github.com/cloudfoundry/cli/testhelpers/configuration"
+	testreq "github.com/cloudfoundry/cli/testhelpers/requirements"
+	testterm "github.com/cloudfoundry/cli/testhelpers/terminal"
+
+	. "github.com/cloudfoundry/cli/cf/commands/servicekey"
+	. "github.com/cloudfoundry/cli/testhelpers/matchers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("service-key command", func() {
+	var (
+		ui                  *testterm.FakeUI
+		config              core_config.Repository
+		cmd                 ServiceKey
+		requirementsFactory *testreq.FakeReqFactory
+		serviceRepo         *testapi.FakeServiceRepo
+		serviceKeyRepo      *testapi.FakeServiceKeyRepo
+	)
+
+	BeforeEach(func() {
+		ui = &testterm.FakeUI{}
+		config = testconfig.NewRepositoryWithDefaults()
+		serviceRepo = &testapi.FakeServiceRepo{}
+		serviceInstance := models.ServiceInstance{}
+		serviceInstance.Guid = "fake-service-instance-guid"
+		serviceRepo.FindInstanceByNameMap = generic.NewMap()
+		serviceRepo.FindInstanceByNameMap.Set("fake-service-instance", serviceInstance)
+		serviceKeyRepo = testapi.NewFakeServiceKeyRepo()
+		cmd = NewGetServiceKey(ui, config, serviceRepo, serviceKeyRepo)
+		requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, TargetedSpaceSuccess: true, ServiceInstanceNotFound: false}
+	})
+
+	var callGetServiceKey = func(args []string) bool {
+		return testcmd.RunCommand(cmd, args, requirementsFactory)
+	}
+
+	Describe("requirements", func() {
+		It("fails when not logged in", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: false}
+			Expect(callGetServiceKey([]string{"fake-service-key-name"})).To(BeFalse())
+		})
+
+		It("requires two arguments and one option to run", func() {
```

Comment:
This should read "requires two arguments to run", the `--guid` flag is an option and is not required to run.

And to actually validate this test, add a new `Expect` to show that providing 2 arguments will indeed pass requirement "BeTrue()"